#!/bin/sh /usr/share/dpatch/dpatch-run
## 02-amd64.dpatch by Robert Millan <rmh@aybabtu.com>
##
## DP: support for building on amd64 and others

@DPATCH@
diff -urNad invaders-1.0.0~/compile.sh invaders-1.0.0/compile.sh
--- invaders-1.0.0~/compile.sh	2001-05-21 13:26:46.000000000 -0700
+++ invaders-1.0.0/compile.sh	2009-10-02 14:20:30.000000000 -0700
@@ -3,8 +3,8 @@
 set -e
 set -v
 
-CFLAGS="-fno-builtin -nostdinc -O2 -I. -Wall -Werror"
-LDFLAGS="-nostdlib -Wl,-N -Wl,-Ttext -Wl,100000"
+CFLAGS="-fno-builtin -nostdinc -O2 -I. -Wall -Werror -m32 -nostdlib"
+LDFLAGS="-Wl,-N -Wl,-Ttext -Wl,100000 -m32"
 gcc $CFLAGS -c keyboard.c
 gcc $CFLAGS -c delay.c
 gcc $CFLAGS -c common.c
@@ -14,5 +14,5 @@
 gcc $CFLAGS -c video.c
 gcc $CFLAGS -c memory.c
 gcc $CFLAGS -c boot.S
-gcc $LDFLAGS -o invaders.exec boot.o kernel.o keyboard.o video.o game.o sound.o delay.o common.o memory.o
+gcc $CFLAGS $LDFLAGS -o invaders.exec boot.o kernel.o keyboard.o video.o game.o sound.o delay.o common.o memory.o
 objcopy -O binary invaders.exec invaders
